source/MaterialXRuntime/Private/PvtPrim.cpp

```diff
@@ -224,7 +216,8 @@ RtToken PvtPrim::makeUniqueChildName(const RtToken& name) const
 
     // Check if there is another child with this name.
     const PvtPrim* otherChild = getChild(name);
-    if (otherChild)
+    const PvtAttribute* otherAttr = getAttribute(name);
```

Comment:
These are the changes @bernardkwok and I originally discussed. Unfortunately a bunch of other changes were needed as well in this review.